ts/src/profiler.ts

```diff
@@ -13,47 +13,387 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import * as http from 'http';
 import * as path from 'path';
+import * as pify from 'pify';
+import * as zlib from 'zlib';
 
-import {AuthenticationConfig, Common, Logger, Service, ServiceConfig} from '../third_party/types/common-types';
-
-import {Config, ProfilerConfig} from './config';
+import {perftools} from '../../proto/profile';
+import {AuthenticationConfig, Common, Logger, Service, ServiceConfig, ServiceObject, ServiceObjectConfig} from '../third_party/types/common-types';
+import {ProfilerConfig} from './config';
 import {HeapProfiler} from './profilers/heap-profiler';
 import {TimeProfiler} from './profilers/time-profiler';
 
+export const common: Common = require('@google-cloud/common');
 const pjson = require('../../package.json');
-const common: Common = require('@google-cloud/common');
+const API = 'https://cloudprofiler.googleapis.com/v2';
+const zoneNameLabel = 'zone';
+const versionLabel = 'version';
+const instanceLabelName = 'instance';
+const scope = 'https://www.googleapis.com/auth/monitoring.write';
+const gzip = pify(zlib.gzip);
+
+enum ProfileTypes {
+  Wall = 'WALL',
+  Heap = 'HEAP'
+}
+
+/**
+ * Parses string containing a time duration, and returns this duration in ms.
+ * If duration cannot be parsed, returns undefined.
+ *
+ * Public for testing.
+ */
+export function parseDurationMillis(durationString: string): number|undefined {
```

Comment:
nit: prefer `null` over `undefined` for the sentinel value. Generally I equate `undefined` with 'lack of value' and `null` with 'the nothing value'. Somewhat relevant, if a bit obtuse, [read](http://thecodelesscode.com/case/6).